test/scripts/run_cuda_memcheck.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+"""This script runs cuda-memcheck on the specified unit test. Each test case
+is run in its isolated process with a timeout so that:
+1) different test cases won't influence each other, and
+2) in case of hang, the script would still finish in a finite amount of time.
+The output will be written to a log file result.log
+
+Example usage:
+    python run_cuda_memcheck.py ../test_torch.py 600
+
+Note that running cuda-memcheck could be very slow.
+"""
+
+import asyncio
+import torch
+import multiprocessing
+import argparse
+import subprocess
+import tqdm
+import re
+
+ALL_TESTS = []
+NUM_PROCESSES = multiprocessing.cpu_count()
```

Comment:
You should let users specify the number of processes and override this value.

Reply:
fixed